app/controllers/application_controller.rb

```diff
@@ -1,10 +1,19 @@
 class ApplicationController < ActionController::Base
   protect_from_forgery with: :exception
 
-  helper_method :cart
+  helper_method :cart,
+                :current_user,
+                :current_user?
 
   def cart
     @cart ||= Cart.new(session[:cart] ||= Hash.new(0))
   end
 
+  def current_user
+    @current_user ||= User.find(session[:user_id]) if session[:user_id]
+  end
+
+  def current_user?
+    current_user && current_user.user?
```

Comment:
This is confusing. Should we rename a registered user as default?

Reply:
Based on the user stories, a registered user is the default user. In our setup, we named a default user role just user, and it's a bit confusing for sure. I would agree that we should come up with a definitive term for a registered (default) user.

Reply:
rename to registered_user to keep things clear